platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/AsyncPreConsensusEventWriter.java

```diff
@@ -0,0 +1,190 @@
+    @Override
+    public void writeEvent(final EventImpl event) throws InterruptedException {
+        if (event.getStreamSequenceNumber() == EventImpl.NO_STREAM_SEQUENCE_NUMBER
+                || event.getStreamSequenceNumber() == EventImpl.STALE_EVENT_STREAM_SEQUENCE_NUMBER) {
+            throw new IllegalStateException("Event must have a valid stream sequence number");
+        }
+        eventInserter.put(event);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setMinimumGenerationNonAncient(final long minimumGenerationNonAncient) throws InterruptedException {
+        minimumGenerationNonAncientInserter.put(minimumGenerationNonAncient);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setMinimumGenerationToStore(final long minimumGenerationToStore) {
+        writer.setMinimumGenerationToStore(minimumGenerationToStore);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isEventDurable(final EventImpl event) {
+        return writer.isEventDurable(event);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void waitUntilDurable(final EventImpl event) throws InterruptedException {
+        writer.waitUntilDurable(event);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean waitUntilDurable(final EventImpl event, final Duration timeToWait) throws InterruptedException {
+        return writer.waitUntilDurable(event, timeToWait);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void requestFlush(final EventImpl event) {
+        writer.requestFlush(event);
+    }
+
+    /**
+     * Pass a minimum generation non-ancient to the wrapped writer.
+     */
+    private void setMinimumGenerationNonAncientHandler(final Long minimumGenerationNonAncient) {
+        try {
+            writer.setMinimumGenerationNonAncient(minimumGenerationNonAncient);
+        } catch (final InterruptedException e) {
+            // Unless we do something silly like wrapping an asynchronous writer inside another asynchronous writer,
+            // this should never throw an InterruptedException.
+            logger.error(
+                    EXCEPTION.getMarker(),
+                    "interrupted while attempting to call setMinimumGenerationNonAncient on writer");
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    /**
+     * Pass an event to the wrapped writer.
+     */
+    private void addEventHandler(final EventImpl event) {
+        try {
+            writer.writeEvent(event);
+        } catch (final InterruptedException e) {
+            // Unless we do something silly like wrapping an asynchronous writer inside another asynchronous writer,
+            // this should never throw an InterruptedException.
+            logger.error(EXCEPTION.getMarker(), "interrupted while attempting to call addEvent on writer");
+            Thread.currentThread().interrupt();
```

Comment:
why?

Reply:
The synchronous implementation we are calling into doesn't throw this exception type, but java forces me to catch it because of the interface.